### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by rplopes**
Update dependencies to their latest version. Includes an update to the new package we use to generate changelogs.

The code for `defaultsDeep()` was significantly changed, copied from @Americas's implementation, in order to keep its behaviour after the major lodash update.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,18 +26,18 @@
...
-    "github-changes": "0.0.16",
-    "mocha": "^2.0.1",
-    "should": "^4.2.1"
+    "@uphold/github-changelog-generator": "^0.7.0",
+    "mocha": "^5.2.0",
+    "should": "^13.2.3"
   },
   "engines": {
-    "node": ">= 0.10"
+    "node": ">=4.0.0"
```
Comment (Americas) [inline package.json:40]: Version 4 is no longer being maintained right? Maybe we should bump it to 6.

**Review by Americas — CHANGES_REQUESTED** [review]
Maybe update the readme too, remove (or update) the status buttons.

File: package.json (hunk 1 above)
Comment (rplopes) [reply package.json:40]: I'd keep it at 4, if that's what the current version of lodash is supporting.

**Comment by rplopes** [comment]
Good point. Updated.

**Review by Americas — APPROVED** [review]

**Pull request merged**